### PR TITLE
prevent POP actions to push the url into the history.

### PR DESCRIPTION
**Pull request by lluistfc**
When using your component with react (v16) + react-router (v4) and react-redux (v5) i've encontered an issue when doing a back from the browser.

If we start on page A and go to page B -> push event.
Once we are on page B we go to page C -> push event.
Once in page C we do a browser back, we end on page B -> pop event. but page B gets pushed onto history.
Back on page B we do a browser back, we end on page C

That behaviour is caused by:
```php
if (pathnameInHistory !== pathnameInStore || searchInHistory !== searchInStore || hashInHistory !== hashInStore) {
  this.inTimeTravelling = true
  // Update history's location to match store's location
  props.history.push({
    pathname: pathnameInStore,
    search: searchInStore,
    hash: hashInStore,
  })
}
```
on [ConnectedRouter.js#L38](https://github.com/supasate/connected-react-router/blob/master/src/ConnectedRouter.js#L38)

By adding another condition to check if history action is "PUSH" i've managed to avoid this behaviour.

```php
if (props.history.action === 'PUSH' && (pathnameInHistory !== pathnameInStore || searchInHistory !== searchInStore || hashInHistory !== hashInStore)) {
```

### CODE REVIEW

**Comment by supasate** [comment]
Thanks @lluistfc. Last time I looked into this and found some caveats. Sadly, I forget it now. Let me take a closer look into it again.

**Comment by lluistfc** [comment]
Hello @supasate. Any news on this issue?

**Comment by vorlov** [comment]
pls check this asap, its a critical issue as the page is rendered twice on browser back action

**Comment by quoo** [comment]
Any updates on this?

**Comment by supasate** [comment]
Just revisited this issue. 

I'm not sure that this fix is addressing general cases of clicking back button or only during the time traveling.

In the case of clicking the browser back button outside of time traveling, I cannot reproduce this behavior -> "Once in page C we do a browser back, we end on page B -> pop event. but page B gets pushed onto history."

From the codebase, page B will be pushed again only if it's in time traveling, i.e. the location in history and the location in redux store are different. By clicking back outside of time traveling, both locations are synced.

**Comment by supasate** [comment]
@vorlov the page is rendered twice is a separate issue and this PR doesn't fix that. This issue is reported in #141. Let discuss in that issue.

**Comment by pcs2112** [comment]
Critical issue. @lluistfc update fixes navigation issues using the browser's back button.

**Comment by reallistic** [comment]
Here is a (hopefully) a full explanation of what is going on with the code in question:

When using the redux dev tools to "time travel" (I.e. replay previous versions of your state) the browser history can become out of sync. The original code was designed to remedy this via the following _intended_ logic:
```
Developer is on Page B.
Developer uses redux time travel to revert state to a point when Page A was active.
ConnectedRouter.js detects the mismatch of router state (changed) and history state (unchanged).
Page A is pushed onto window stack.
```

However, the unintended consequence is the following:
```
User is on Page B.
User hits the browser back button to go back to Page A.
A component mount on Page A fires an action that causes state to change.
ConnectedRouter.js detects the mismatch of router state (unchanged) and history state (updated to be Page A).
Page B is pushed onto window stack.
The LOCATION_CHANGED event fires due to the history listener POP'ing to Page A.
The router reducer updates the store.
ConnectedRouter.js detects the mismatch of router state (updated to Page A) and history state (on Page B).
Page A is pushed onto window stack.
```

**Comment by siddhant91** [comment]
@supasate  @lluistfc  Is there any update on when this will be merged and published.

**Comment by ymaz** [comment]
any updates on that?

File: src/ConnectedRouter.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ const createConnectedRouter = (structure) => {
         } = props.history.location
 
         // If we do time travelling, the location in store is changed but location in history is not changed
-        if (pathnameInHistory !== pathnameInStore || searchInHistory !== searchInStore || hashInHistory !== hashInStore) {
+        if (props.history.action === 'PUSH' && (pathnameInHistory !== pathnameInStore || searchInHistory !== searchInStore || hashInHistory !== hashInStore)) {
```
Comment (jancama2) [inline src/ConnectedRouter.js]: Would be nice to have here some detailed comment to explain why it is important to check the action.

Comment (jancama2) [reply src/ConnectedRouter.js]: Could we test it to be sure it wont break the next time?

**Comment by pybuche** [comment]
Hi ! Any updates on that ?

**Comment by JesusADS** [comment]
can anyone with permissions fix the .gitignore conflict?

**Comment by jwineman** [comment]
@jancama2 if I PR this same fix with tests and no `.gitignore` changes would you accept it?

**Comment by roryp2** [comment]
Is this going to be updated and merged? This fixes https://github.com/supasate/connected-react-router/issues/262

**Comment by supasate** [comment]
Sorry all for keeping this PR for a long time! TBH, I still don't have time to think about it thoroughly, but, it seems to fix many issues listed above. So, I'm merging it first. If anyone finds issues after merging, please let me know!

**Comment by pybuche** [comment]
Great, thanks ! Do you know when this will be released?

**Pull request merged**